Services/ChatHub.cs

```diff
@@ -1,34 +1,49 @@
 ﻿using Microsoft.AspNetCore.SignalR;
 using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
 using System.Collections.Generic;
-using System.Security.Cryptography;
+using System.Linq;
+using Karma.Repositories;
+using System.IO;
+using System;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
 
 namespace Karma.Services
 {
+    [Authorize]
     public class ChatHub : Hub
     {
-        public Task SendMessage(string user, string message)
+        //TODO: Find a better way to do DI
+        private IMessageService _messageService = new MessageService(new MessageRepository(Path.Combine("data", "messages")));
+       
+        public Task SendMessageAsync(string user, string message)
         {
             return Clients.All.SendAsync("ReceiveMessage", user, message);
         }
 
-        public Task SendMessageToCaller(string user, string message)
+        public Task SendMessageToCallerAsync(string user, string message)
         {
             return Clients.Caller.SendAsync("ReceiveMessageToCaller", user, message);
         }
 
-        public Task SendGroupMessage(string groupId, string user, string message)
-        {
+        public Task SendGroupMessageAsync(string groupId, string user, string message)
+        {   
+            _messageService.AddMessage(message, Context.UserIdentifier, groupId);
+            _messageService.SaveMessages(groupId);
+
             return Clients.Group(groupId).SendAsync("ReceiveGroupMessage", user, message);
         }
 
-        public async Task AddToGroup(string groupId)
```

Comment:
Currently, chat doesn't work for me. 
![image](https://user-images.githubusercontent.com/90030313/141349371-8ddedd93-4827-4380-8111-ecc2c23a943f.png)
I believe this is due to the fact that you changed method names, but didn't modify arguments of SendAsync method instances in frontend and backend.

Reply:
Fixed!